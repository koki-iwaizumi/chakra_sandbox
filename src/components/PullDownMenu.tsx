import { FC } from "react";
import {
  Menu,
  MenuButton,
  MenuList,
  MenuItem,
  Button,
  SpaceProps,
} from "@chakra-ui/react";
import { ChevronDownIcon } from "@chakra-ui/icons";

export interface Props {
  text: string;
  m?: SpaceProps["m"];
}

export const PullDownMenu: FC<Props> = (props: Props) => {
  return (
    <Menu>
      <MenuButton m={props.m} as={Button} rightIcon={<ChevronDownIcon />}>
        {`${props.text}:${props.m}`}
      </MenuButton>
      <MenuList>
        <MenuItem>Download</MenuItem>
        <MenuItem>Create a Copy</MenuItem>
        <MenuItem>Mark as Draft</MenuItem>
        <MenuItem>Delete</MenuItem>
        <MenuItem>Attend a Workshop</MenuItem>
      </MenuList>
    </Menu>
  );
};
