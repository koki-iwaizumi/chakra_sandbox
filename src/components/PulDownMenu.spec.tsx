import "@testing-library/jest-dom";
import { render, screen, logRoles, getByText } from "@testing-library/react";
import { PullDownMenu } from "./PullDownMenu";
import userEvent from "@testing-library/user-event";

describe("PullDownMenu", () => {
  it("should foo", () => {
    const { asFragment } = render(<PullDownMenu text="menu" m={1} />);
    // expect(asFragment()).toMatchSnapshot();
  });

  it("should bar", () => {
    const { container } = render(<PullDownMenu text="menu" m={1} />);
    // logRoles(container);
    userEvent.click(screen.getByRole("button", { name: "menu:1" }));
    // logRoles(container);

    console.log(screen.getAllByRole("menuitem", { hidden: true }).length);
    // userEvent.click(screen.getByRole("menuitem"));
  });
});
