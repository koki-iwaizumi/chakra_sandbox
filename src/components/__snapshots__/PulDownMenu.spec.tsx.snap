// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullDownMenu should foo 1`] = `
<DocumentFragment>
  <button
    aria-controls="menu-list-8605045946-2"
    aria-expanded="false"
    aria-haspopup="menu"
    class="chakra-button chakra-menu__menu-button css-i7psh6"
    id="menu-button-8605045946-2"
    type="button"
  >
    <span
      class="css-xl71ch"
    >
      menu:1
    </span>
    <span
      class="chakra-button__icon css-1hzyiq5"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          fill="currentColor"
        />
      </svg>
    </span>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-sjt5nk"
      id="menu-list-8605045946-2"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <button
        class="chakra-menu__menuitem css-1k756t9"
        data-index="0"
        id="menu-list-8605045946-2-menuitem-8605045946-3"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Download
      </button>
      <button
        class="chakra-menu__menuitem css-1k756t9"
        data-index="1"
        id="menu-list-8605045946-2-menuitem-8605045946-4"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Create a Copy
      </button>
      <button
        class="chakra-menu__menuitem css-1k756t9"
        data-index="2"
        id="menu-list-8605045946-2-menuitem-8605045946-5"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Mark as Draft
      </button>
      <button
        class="chakra-menu__menuitem css-1k756t9"
        data-index="3"
        id="menu-list-8605045946-2-menuitem-8605045946-6"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Delete
      </button>
      <button
        class="chakra-menu__menuitem css-1k756t9"
        data-index="4"
        id="menu-list-8605045946-2-menuitem-8605045946-7"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Attend a Workshop
      </button>
    </div>
  </div>
</DocumentFragment>
`;
